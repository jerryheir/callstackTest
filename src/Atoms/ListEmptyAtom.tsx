import React from 'react';
import { View, Image, Text, ActivityIndicator } from 'react-native'; 
import { styles } from "../Styles";
import { colors } from '../Styles/Colors';

const ListEmptyAtom = (props: any) => {
    const { loading } = props;
    return (
        <View style={styles.container404}>
            {
                (loading === false) && 
                <View>
                    <Image 
                    source={require('../Images/git-404.png')} 
                    style={styles.image404Style}
                    />
                    <Text style={[styles.text404, { fontSize: 42 }]}>Not Found</Text>
                    <Text style={styles.text404}>404</Text>
                </View>
            }
            {
                (loading === true) &&
                <View style={styles.loadingView}>
                    <ActivityIndicator size="large" color={colors.white} />
                </View>
            }
        </View>
    )
}

export default ListEmptyAtom;
