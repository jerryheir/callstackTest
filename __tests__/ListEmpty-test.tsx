/**
 * @format
 */

import 'react-native';
import React from 'react';
import ListEmptyAtom from '../src/Atoms/ListEmptyAtom';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('emptylist correctly', () => {
    renderer.create(<ListEmptyAtom loading={true} />);
});

test('emptylist correctly', () => {
    const tree = renderer.create(<ListEmptyAtom loading={true} />).toJSON();
    expect(tree).toMatchSnapshot();
});