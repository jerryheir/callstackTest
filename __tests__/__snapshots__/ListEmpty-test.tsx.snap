// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emptylist correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "height": "60%",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "borderRadius": 8,
        "height": 80,
        "justifyContent": "center",
        "marginTop": 120,
        "width": 80,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#FFF"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
